pkg/datapath/linux/config/config.go

```diff
@@ -1049,30 +1033,23 @@ func (h *HeaderfileWriter) writeStaticData(devices []string, fw io.Writer, e dat
 }
 
 // WriteEndpointConfig writes the BPF configuration for the endpoint to a writer.
-func (h *HeaderfileWriter) WriteEndpointConfig(w io.Writer, e datapath.EndpointConfiguration) error {
+func (h *HeaderfileWriter) WriteEndpointConfig(w io.Writer, cfg *datapath.LocalNodeConfiguration, e datapath.EndpointConfiguration) error {
 	fw := bufio.NewWriter(w)
 
-	var (
-		nativeDevices []*tables.Device
-		deviceNames   []string
-	)
-	if h.db != nil && h.devices != nil {
-		nativeDevices, _ = tables.SelectedDevices(h.devices, h.db.ReadTxn())
-		deviceNames = tables.DeviceNames(nativeDevices)
-	}
+	deviceNames := cfg.DeviceNames()
 
 	// Add cilium_wg0 if necessary.
 	if option.Config.NeedBPFHostOnWireGuardDevice() {
-		deviceNames = append(deviceNames, wgtypes.IfaceName)
+		deviceNames = append(slices.Clone(deviceNames), wgtypes.IfaceName)
```

Comment:
It's not strictly necessary given how `deviceNames` is constructed, but since it's cheap to clone I'm just being careful here and making this not rely on the details of how we got `deviceNames`.